interfaces/cython/cantera/test/test_reactor.py

```diff
@@ -1542,3 +1542,39 @@ def test_ConstPressureReactor(self):
         self.assertEqual(states.X[-2], 1)
         for i in range(3,7):
             self.assertNear(states.T[i], states.T[2])
+
+
+class ImplicitSurfChemTest(utilities.CanteraTest):
+    def test_advance_coverages_parameters(self):
+        # create gas and interface
+        gas1 = ct.Solution('ptcombust.xml', 'gas')
+        surf = ct.Interface('ptcombust.xml', 'Pt_surf', [gas1])
+
+        # first, test max step size & max steps
+        dt = 1.0
+        max_steps = 10
+        max_step_size = dt / (max_steps + 1)
+        # this should throw an error, as we can't reach dt
+        try:
+            surf.advance_coverages(
+                dt=dt, max_step_size=max_step_size, max_steps=max_steps)
+            self.fail()
+        except ct.CanteraError:
+            pass
+
+        # next, run with different tolerances
+        gas1 = ct.Solution('ptcombust.xml', 'gas')
+        surf = ct.Interface('ptcombust.xml', 'Pt_surf', [gas1])
+        surf.coverages = 'O(S):0.1, PT(S):0.5, H(S):0.4'
+        gas1.TP = surf.TP
+
+        surf.advance_coverages(dt=dt, rtol=1e-5, atol=1e-12)
+        cov = surf.coverages[:]
+
+        surf.coverages = 'O(S):0.1, PT(S):0.5, H(S):0.4'
+        gas1.TP = surf.TP
+        surf.advance_coverages(dt=dt, rtol=1e-7, atol=1e-14)
+
+        # check that the solutions are similar, but not identical
+        self.assertArrayNear(cov, surf.coverages)
+        assert any(cov != surf.coverages)
```

Comment:
All of the test assertions should be methods of the test class, rather than plain `assert` statements, in order to get good error messages out of them.